mlir/include/mlir/Dialect/LLVMIR/LLVMOps.td

```diff
@@ -1575,6 +1575,44 @@ def LLVM_AliasOp : LLVM_Op<"mlir.alias",
   let hasRegionVerifier = 1;
 }
 
+def LLVM_DSOLocalEquivalentOp : LLVM_Op<"dso_local_equivalent",
+    [Pure, ConstantLike, DeclareOpInterfaceMethods<SymbolUserOpInterface>]> {
+  let arguments = (ins FlatSymbolRefAttr:$function_name);
```

Comment:
If we use inheritance using the new attribute here could make sense. Or was it printing strangely?

Reply:
That doesn't work, we get some type issues that we don't get with `FlatSymbolRefAttr` (it probably requires teaching something special to tablegen?):
```
error: undefined type for argument #0
def LLVM_DSOLocalEquivalentOp : LLVM_Op<"dso_local_equivalent",
```